### PR TITLE
Support additional load paths

### PR DESCRIPTION
**Pull request by squrious**
Hi there, 

This is a proposal implementation for #28

### CODE REVIEW

**Comment by squrious** [comment]
Rebased to get the fix for `FunctionalTest`

File: src/SassBuilder.php [hunk 1]
```diff
@@ -26,6 +26,7 @@ public function __construct(
         private readonly string $projectRootDir,
         private readonly ?string $binaryPath,
         private readonly bool $embedSourcemap,
+        private readonly array $loadPaths,
```
Comment (bocharsky-bw) [inline src/SassBuilder.php:29]: I wonder if it will lead to BC breaks if we release this feature as a minor version bump?

**Review by bocharsky-bw — COMMENTED** [review]
This looks good to me, thanks for describing this feature in the docs!

File: src/SassBuilder.php (hunk 1 above)
Comment (squrious) [reply src/SassBuilder.php:29]: Hmm yes it could. Would default to an empty array be enough to handle this?

Comment (bocharsky-bw) [reply src/SassBuilder.php:29]: Yeah, I think setting it to an empty array should do the trick. Probably we could add a docblock comment explaining it's default to an empty array just for BC

Comment (squrious) [reply src/SassBuilder.php]: I added the default value and related docblock entry

File: src/DependencyInjection/SymfonycastsSassExtension.php [hunk 2]
```diff
@@ -77,6 +78,12 @@ public function getConfigTreeBuilder(): TreeBuilder
                         ->end()
                     ->defaultValue(['%kernel.project_dir%/assets/styles/app.scss'])
                 ->end()
+                ->arrayNode('load_paths')
+                    ->info('Additional paths to load')
+                    ->scalarPrototype()
+                        ->end()
+                    ->defaultValue([])
```
Comment (stof) [inline src/DependencyInjection/SymfonycastsSassExtension.php:85]: this is useless. prototyped nodes always have a default value automatically.

File: src/DependencyInjection/SymfonycastsSassExtension.php [hunk 3]
```diff
@@ -77,6 +78,12 @@ public function getConfigTreeBuilder(): TreeBuilder
                         ->end()
                     ->defaultValue(['%kernel.project_dir%/assets/styles/app.scss'])
                 ->end()
+                ->arrayNode('load_paths')
```
Comment (stof) [inline src/DependencyInjection/SymfonycastsSassExtension.php:81]: To make it possible for people wanting it to use XML config files, you should add `->fixXmlConfig('load_path')` on the parent node (which is the root node)

File: src/DependencyInjection/SymfonycastsSassExtension.php [hunk 4]
```diff
@@ -77,6 +78,12 @@ public function getConfigTreeBuilder(): TreeBuilder
...
+                    ->info('Additional paths to load')
```
Comment (stof) [inline src/DependencyInjection/SymfonycastsSassExtension.php:82]: those are `Additional load paths`, not `Additional paths to load`

File: doc/index.rst [hunk 5]
```diff
@@ -151,3 +151,23 @@ This bundle already installed for you the right binary. However, if you already
 
     symfonycasts_sass:
         binary: 'node_modules/.bin/sass'
+
+Register additional load paths
```
Comment (stof) [inline doc/index.rst:155]: This section of the doc should probably have a link to https://sass-lang.com/documentation/at-rules/use/#finding-the-module

File: src/DependencyInjection/SymfonycastsSassExtension.php (hunk 2 above)
Comment (squrious) [reply src/DependencyInjection/SymfonycastsSassExtension.php]: Fixed

File: src/DependencyInjection/SymfonycastsSassExtension.php (hunk 3 above)
Comment (squrious) [reply src/DependencyInjection/SymfonycastsSassExtension.php:82]: Fixed

File: src/DependencyInjection/SymfonycastsSassExtension.php (hunk 4 above)
Comment (squrious) [reply src/DependencyInjection/SymfonycastsSassExtension.php]: Fixed

File: doc/index.rst (hunk 5 above)
Comment (squrious) [reply doc/index.rst:155]: Added a link in the documentation.

File: src/DependencyInjection/SymfonycastsSassExtension.php [hunk 6]
```diff
@@ -77,6 +78,12 @@ public function getConfigTreeBuilder(): TreeBuilder
                         ->end()
                     ->defaultValue(['%kernel.project_dir%/assets/styles/app.scss'])
                 ->end()
+                ->fixXmlConfig('load_path')
```
Comment (stof) [inline src/DependencyInjection/SymfonycastsSassExtension.php:81]: This is broken. `fixXmlConfig` is a method of the node, not of the NodeBuilder returned by `children()`

Comment (squrious) [reply src/DependencyInjection/SymfonycastsSassExtension.php:81]: Oh sorry I missed that. Also, should `root_sass` be fixed too?

Comment (stof) [reply src/DependencyInjection/SymfonycastsSassExtension.php:81]: `root_sass` will be hard to handle as it is not a plural name for which we might make the repeated XML tag use the singular as name...

Comment (squrious) [reply src/DependencyInjection/SymfonycastsSassExtension.php]: Alright. I just fixed my fix.

File: doc/index.rst [hunk 7]
```diff
@@ -236,3 +239,24 @@ This bundle already installed for you the right binary. However, if you already
...
+-------------------------------
```
Comment (bocharsky-bw) [inline doc/index.rst:244]:
```suggestion
Register Additional Load Paths
------------------------------
```

**Review by bocharsky-bw — COMMENTED** [review]
This looks good to me, but I'm not sure how we can rebuild it to see if tests pass.

**Comment by squrious** [comment]
> This looks good to me, but I'm not sure how we can rebuild it to see if tests pass.

Hi! Is there something I can do here to help?

**Review by bocharsky-bw — APPROVED** [review]
This looks good to me. Also, I don't see any further comments on it, so I will consider it as ready.

Thank you for working on it!

**Pull request merged**